packages/ionic/src/commands/capacitor/base.ts

```diff
@@ -0,0 +1,47 @@
+import { CommandInstanceInfo, IShellRunOptions } from '@ionic/cli-utils';
+import { Command } from '@ionic/cli-utils/lib/command';
+import { FatalException } from '@ionic/cli-utils/lib/errors';
+import { runCommand } from '@ionic/cli-utils/lib/executor';
+
+export abstract class CapacitorCommand extends Command {
+  async checkCapacitor(runinfo: CommandInstanceInfo) {
+    const project = await this.env.project.load();
+
+    if (project.integrations.capacitor && project.integrations.capacitor.enabled === false) {
+      return;
+    }
+    if (!project.integrations.capacitor) {
+      await runCommand(runinfo, ['integrations', 'enable', 'capacitor']);
+    }
+  }
+
+  async preRunChecks(runinfo: CommandInstanceInfo) {
+    runinfo;
+  }
+
+  async runCapacitor(argList: string[], { fatalOnNotFound = false, truncateErrorOutput = 5000, ...options }: IShellRunOptions = {}): Promise<void> {
+    const { ERROR_SHELL_COMMAND_NOT_FOUND } = await import('@ionic/cli-utils/lib/shell');
+    try {
+      this.env.close();
+      await this.env.shell.run('npx', ['capacitor'].concat(argList), { fatalOnNotFound, truncateErrorOutput, stdio: 'inherit', ...options });
```

Comment:
You shouldn't need to use `npx`, if you prefer. The `.run()` utility adds `node_modules/.bin` to `PATH` in the same way npx does.

Reply:
Cool should be more stable to just call the binary directly